src/components/card/card.tsx

```diff
@@ -40,32 +38,47 @@ export const Card = ({offer, onSelect, isActive, onOfferClickHandler, isNear}: C
     setIsUpdating(false);
   };
 
+  const onOfferClickHandler = ({evt}: OfferClickHandlerProps) => {
+    evt.stopPropagation();
+    dispatch(fetchCurrentOfferAction(offer));
+    dispatch(fetchCommentsAction(offer));
+    dispatch(fetchNearestOfferAction(offer));
+  };
+
   return (
     <article className={cn(
       'place-card',
-      {'cities__card': !isNear},
-      {'near-places__card': isNear}
+      {'cities__card': cardType === CardType.MAIN},
+      {'near-places__card': cardType === CardType.NEAR},
+      {'favorites__card': cardType === CardType.FAVORITES}
     )}
-    onMouseEnter={() => onSelect(offer.id)}
-    onMouseLeave={() => onSelect(null)}
-    onClick={(evt) => onOfferClickHandler({offer, evt})}
-    data-active={isActive ? 'true' : undefined}
+    onMouseEnter={() => dispatch(changeActiveOfferId(offer.id))}
+    onMouseLeave={() => dispatch(changeActiveOfferId(null))}
+    onClick={(evt) => onOfferClickHandler({evt})}
     >
       { offer.isPremium ?
         <div className="place-card__mark">
           <span>Premium</span>
         </div> : '' }
       <div className={cn(
         'place-card__image-wrapper',
-        {'cities__image-wrapper': !isNear},
-        {'near-places__image-wrapper': isNear}
+        {'cities__image-wrapper': cardType === CardType.MAIN},
+        {'near-places__image-wrapper': cardType === CardType.NEAR},
+        {'favorites__image-wrapper': cardType === CardType.FAVORITES}
       )}
       >
         <Link to={`/offer/${offer.id}`}>
-          <img className="place-card__image" src={offer.previewImage} width="260" height="200" alt="Place image"/>
+          {cardType === CardType.FAVORITES ?
+            <img className="place-card__image" src={offer.previewImage} width="150" height="110" alt="Place image"/>
+            :
+            <img className="place-card__image" src={offer.previewImage} width="260" height="200" alt="Place image"/>}
```

Comment:
Мемоизация здесь, конечно, не очень нужна. Но, как пример, можно воткнуть :-) 
Ну и в целом порефакторить это дело

```js
import React, { memo } from 'react';
import cn from 'classnames';

// Memoize the component to avoid unnecessary re-renders
const PlaceCardImage = memo(({ cardType, offer }) => {
  const isFavorites = cardType === CardType.FAVORITES;
  
  const imageAttributes = {
    src: offer.previewImage,
    alt: "Place image",
    width: isFavorites ? "150" : "260",
    height: isFavorites ? "110" : "200",
  };

  return (
    <img
      className="place-card__image"
      {...imageAttributes}
    />
  );
});


// Usage inside a component
<PlaceCardImage cardType={cardType} offer={offer} />
```